### PR TITLE
test: add integration test

### PR DESCRIPTION
**Pull request by ibrizsabin**
### Commit Message

```bash
type: commit message
```

see the [guidelines](https://github.com/icon-project/community/blob/main/guidelines/technical/software-development-guidelines.md#commit-messages) for commit messages.

### Changelog Entry

```bash
version: <log entry>
```

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have documented my code in accordance with the [documentation guidelines](https://github.com/icon-project/community/blob/main/guidelines/technical/software-development-guidelines.md#documentation)
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have run the unit tests
- [ ] I only have one commit (if not, squash them into one commit).
- [ ] I have a descriptive commit message that adheres to the [commit message guidelines](https://github.com/icon-project/community/blob/main/guidelines/technical/software-development-guidelines.md#commit-messages)

> Please review the [CONTRIBUTING.md](/CONTRIBUTING.md) file for detailed contributing guidelines.

### CODE REVIEW

**Review by AntonAndell — CHANGES_REQUESTED** [review]
Would it be possible to also add a test for the response during execution or is that to complicated?

File: contracts/cosmwasm-vm/cw-integration/tests/test_xcall_connection.rs [hunk 1]
```diff
@@ -0,0 +1,360 @@
+mod setup;
```
[the remark this answers is not in the document]
Comment (ibrizsabin) [reply contracts/cosmwasm-vm/cw-integration/tests/test_xcall_connection.rs:1]: ya let me fix that.

File: contracts/cosmwasm-vm/cw-integration/tests/test_xcall.rs [hunk 2]
```diff
@@ -0,0 +1,360 @@
...
+    println!("{result:?}");
+    assert!(result.is_ok());
+    let result = result.unwrap();
+    let event = get_event(&result, "wasm-CallMessageSent").unwrap();
+    println!("{event:?}");
+    assert_eq!(&format!("{nid}/{dapp}"), event.get("to").unwrap());
+}
+
+#[test]
+fn test_rollback_reply() {
```
Comment (AntonAndell) [inline contracts/cosmwasm-vm/cw-integration/tests/test_xcall.rs:298]: What does this test test? I having a hard time following it

Comment (ibrizsabin) [reply contracts/cosmwasm-vm/cw-integration/tests/test_xcall.rs:298]: it is asserting that dapp can send reply when message with rollback is executed and we receive a message in ack with expected parameters.

**Review by AntonAndell — APPROVED** [review]
Looks good

**Pull request merged**